lib/Scanner/ICAPClient.php

```diff
@@ -15,28 +15,29 @@ public function __construct(string $host, int $port) {
 		$this->port = $port;
 	}
 
-	private function connect(): bool {
-		$this->socket = \socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
-
-		if (!\socket_connect($this->socket, $this->host, $this->port)) {
-			return false;
-		}
-
-		# McAfee seems to not properly close the socket once all response bytes are sent to the client
-		# we use a 5 sec time out on receiving data
-		if (!\socket_set_option($this->socket, SOL_SOCKET, SO_RCVTIMEO, ["sec" => 5, "usec" => 0])) {
-			return false;
+	private function connect(): void {
+		$this->writeHandle = @\stream_socket_client(
+			"tcp://{$this->host}:{$this->port}",
+			$errorCode,
+			$errorMessage,
+			5
+		);
+
+		if (!$this->writeHandle) {
+			throw new InitException(
+				\sprintf(
+					'Cannot connect to "tcp://%s:%s": %s (code %d)',
+					$this->host,
+					$this->port,
+					$errorMessage,
```

Comment:
Did you check it works properly? I'm not sure about the scope of the variables so it might always be null...

Reply:
returned by `stream_socket_client` in case of failure:
errno Will be set to the system level error number if connection fails.
errstr Will be set to the system level error message if the connection fails.

https://www.php.net/manual/en/function.stream-socket-client.php

Reply:
Normally you do something like:
```
$errorMessage = null;
stream_socket_client(....., $errorMessage);
echo $errorMessage;
```
The scope of the variable in this case is pretty clear.
However, in your case, I'm not sure if the scope of the `$errorMessage` will be limited to the `stream_socket_client` function, which would mean that the `$errorMessage` variable you're referring here is a new one and won't contain the error message of the `stream_socket_client`

In addition, for me, in the `stream_socket_client` function you're "using" the variable (as a value holder, I know), so I could think that the variable is defined / assigned elsewhere with a different value, so I wouldn't get an undefined variable error if I delete that function and change it for a different one.

Reply:
ok, I swear it works  properly. :crossed_fingers: 
I guess I should send your points about the scope to PHP authors... ;)
as for me the code pretty clear: if I got `false` while opening the socket - I log `errcode` and `errnumber` to debug the reason. If I got anything that is non-false - the socket was opened properly and this is exactly what is written in docs for `stream_socket_client`:
>On success a stream resource is returned which may be used together with the other file functions (such as fgets(), fgetss(), fwrite(), fclose(), and feof()), false on failure.
errno
Will be set to the system level error number if connection fails.
errstr
Will be set to the system level error message if the connection fails.
so I guess the sane usage will always be  as per "Examples" section in https://www.php.net/manual/en/function.stream-socket-client.php
```
$fp = stream_socket_client("tcp://www.example.com:80", $errno, $errstr, 30);
if (!$fp) {
    echo "$errstr ($errno)<br />\n";
} else {
    fwrite($fp, "GET / HTTP/1.0\r\nHost: www.example.com\r\nAccept: */*\r\n\r\n");
    while (!feof($fp)) {
        echo fgets($fp, 1024);
    }
    fclose($fp);
}
```
Also I don't see a point in deletion of random code pieces... Let's delete random letters in code - it's even more funny. :)

Reply:
I mean that the fact `!$fp` is true always mean both `$errstr ($errno)` are defined. This is how PHP authors suggest to process errors for such functions.

Reply:
"You win this time" :stuck_out_tongue: 
The code is small enough to see the whole function, so if we ever need to touch this function I don't think we'll make dumb mistakes.